nautobot/ipam/tests/test_views.py

```diff
@@ -294,6 +424,41 @@ def test_move_ip_addresses_between_namespaces(self):
         created_ip = IPAddress.objects.get(parent__namespace=new_namespace, address=instance.address)
         self.assertEqual(created_ip.parent, new_parent)
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_create_object_warnings(self):
+        self.add_permissions("ipam.add_ipaddress")
+
+        Prefix.objects.create(
+            prefix="192.0.2.0/25",
+            namespace=self.namespace,
+            type=PrefixTypeChoices.TYPE_CONTAINER,
+            status=self.prefix_status,
+        )
+
+        with self.subTest("Creating an IPAddress as a child of a larger Container prefix raises a warning"):
+            self.form_data["address"] = "192.0.2.98/28"
+            response = self.client.post(self._get_url("add"), data=post_data(self.form_data), follow=True)
+            self.assertHttpStatus(response, 200)
+            content = extract_page_body(response.content.decode(response.charset))
+            self.assertIn(
+                "IP address 192.0.2.98/28 currently has prefix 192.0.2.0/25 as its parent, which is a Container. "
+                "This will be considered invalid data in a future release. "
+                "Consider creating an intermediate /28 prefix of type Network to resolve this issue.",
+                strip_tags(content),
+            )
+
+        with self.subTest("Creating an IP as a child of a same-size Container prefix raises a different warning"):
+            self.form_data["address"] = "192.0.2.2/25"
+            response = self.client.post(self._get_url("add"), data=post_data(self.form_data), follow=True)
+            self.assertHttpStatus(response, 200)
+            content = extract_page_body(response.content.decode(response.charset))
+            self.assertIn(
+                "IP address 192.0.2.2/25 currently has prefix 192.0.2.0/25 as its parent, which is a Container. "
+                "This will be considered invalid data in a future release. "
+                "Consider changing the prefix to type Pool to resolve this issue.",
```

Comment:
Why only suggest changing the container to a pool? Would a network not be a suitable type for this situation?

Reply:
See https://github.com/nautobot/nautobot/pull/4036/files#diff-caf8a412a9f30295cacfae888c0fbc1bdc2a7cd2643df55f444414f363113a7bR730-R733 - my reasoning was that the parent of the Container prefix is itself a Network, so changing the Container to a Network would still be invalid data unless you also change the parent Network to a Container. Maybe that's too fiddly though and should be simplified - thoughts?

Reply:
I must have missed that `successful_post` logic. I think it might be better to either change the message to "Network or Pool" on the IPAddress object save or explain why you're suggesting `Pool` or `Network`. Otherwise, the message changes based on conditions that aren't obvious when saving the IPAddress and I think that could be confusing.